plugins/sre-portal/src/components/TerminalComponent/TerminalComponent.tsx

```diff
@@ -0,0 +1,51 @@
+import { makeStyles } from '@material-ui/core';
+import React from 'react';
+import { Terminal } from 'xterm';
+import { FitAddon } from 'xterm-addon-fit';
+import './static/xterm.css';
+
+const useTerminalStyles = makeStyles({
+  root: {
+    height: '100vh',
+  },
+  header: {
+    margin: 0,
+    paddingTop: '0.67em',
+    paddingBottom: '0.67em',
+    textAlign: 'center',
+  },
+  termcontainer: {
+    display: 'flex',
+    width: '100%',
+    height: '100%',
+    alignItems: 'end',
+    flex: 1,
+  },
+  term: {
+    width: '100%',
+  },
+});
+export const TerminalComponent = () => {
+  const fitAddon = new FitAddon();
+  const terminal = new Terminal();
+  terminal.loadAddon(fitAddon);
+  fitAddon.fit();
+  const classes = useTerminalStyles();
+  React.useEffect(() => {
+    const terminalElement = document.getElementById('terminal');
+    if (terminalElement) {
+      terminal.open(terminalElement);
+      terminal.write('$ ');
+    }
+  });
+  return (
+    <>
+      <div className={classes.root}>
+        <h1 className={classes.header}>SRE portal</h1>
+        <div className={classes.termcontainer}>
+          <div className={classes.term} id="terminal" />
+        </div>
+      </div>
+    </>
```

Comment:
This view should be build within the instance only. This plugin should provide a web terminal only.


We don't aim to reimplement prometheus plugin or techdocs plugin in this one, we want to integrate them. Such integration builds on atomic parts - plugins. It should be build within the instance as a new view/page utilizing the existing extensions. This extension is one integrated member not the one integrating it all together. We don't want to introduce plugin bundles like and creating too opinionated plugins.

Reply:
:+1: Didn't realize that we want each part separately and combine them in the instance... Reworked